ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/index.ts

```diff
@@ -53,6 +53,11 @@ const QueueBatchedTestEventPublisher: AzureFunction = async function (
     });
   }
 
+  if (parseSuccessCount < 1) {
+    context.log(`Successfully parsed message count of ${parseSuccessCount} is less than 1; aborting`);
+    return;
+  }

```

Comment:
What happens to messages that are dequeued but then fail to parse? Will they go back on the queue or just disappear?

Reply:
when we dequeue messages, we're not actually deleting them, seems like. that doesn't happen until we call `deleteSuccessfullyParsedMessages`, on the condition that our upload is successful.

additionally, within that method we also skip deleting messages that failed to parse
https://github.com/CDCgov/prime-simplereport/blob/f59fe3971a88d95cac960f8f3bc08dd54763e444/ops/services/app_functions/report_stream_batched_publisher/functions/QueueBatchedReportStreamUploader/lib.ts#L144-L149

so essentially they stay on the queue and we keep trying to parse them until they expire 7 days after initial insertion. [after 5 dequeues a message gets logged to the poison queue](https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue-trigger?tabs=in-process%2Cextensionv5&pivots=programming-language-javascript#poison-messages), but it doesn't look like we do anything with this info. maybe we could have a ticket to alert us if something gets added to the poison queue

Reply:
ignore the last part, that only applies to functions triggered by queues, like our exception handler function

Reply:
Thanks for the explanation 🙌 that makes sense. I actually didn't know about the "poison" queue but that's exactly what I was hoping for